ArduinoFreevalve/ArduinoFreevalve.ino

```diff
@@ -0,0 +1,84 @@
+// An Arduino program for controlling valve timing on a 6.5HP Predator engine from Harbor Freight.
+//
+// License goes here?
+//
+// Wesley Kagan, 2020
+// Ric Allinson, 2020
+
+// Pins assignment.
+const int HALL_MAGNET = 2;
+const int EXHAUST_V = 12;
+const int INTAKE_V = 13;
+
+// Control constants.
+const int DEG_PER_MAGNET = 6; // Number of degrees for per magnet.
+const int ROTATION_TWO = 760/DEG_PER_MAGNET;     // Number of interrupts in the full cycle.
```

Comment:
This should probably be 720 / DEG__PER_MAGNET.  However, a 60-2 wheel has 58 trigger edges over 360 degrees.  Shouldn't there 58*2 -> 116 interrupts per cycle, not 120?

Reply:
Yep, I totally messed up that maths. Why is it 60-2? One for the missing tooth and one for zero based array?

Reply:
I think I made the assumption that he had made a standard 60-2 (60 teeth, 2 missing teeth) wheel.  Re-watching the video it almost looks like a 60-1 wheel.  Either way, it might be easier to define all of these in terms of teeth:

```
    static constexpr int NUM_TEETH = 60;
    static constexpr int NUM_MISSING_TEETH = 2;
    static constexpr int NUM_TRIGGER_EDGES = NUM_TEETH - NUM_MISSING_TEETH;
    static constexpr int DEG_PER_MAGNET = 360 / NUM_TEETH; // Number of degrees for per magnet.
    static constexpr int TRIGGERS_PER_CYCLE = NUM_TRIGGER_EDGES * 2;     // Number of interrupts in the full cycle.
    static constexpr int TRIGGERS_PER_ROTATION = NUM_TRIGGER_EDGES; // Number of interrupts in a half cycle
```.

This way if a change to the trigger wheel design is made you can just update the NUM_TEETH and NUM_MISSING_TEETH.  This would make the intake/exhaust maps a little more complicated.  This could be overcome by creating the maps on an arbitrary grid and then interpolating for the current calculated crank angle.

Reply:
Agreed, rather than teeth I've used STEPS incase there is ever a different way to generate position data. It follows the conventions of stepper motors which implement a similar positioning index system.

I didn't use `constexpr` as I'm not sure it is a standard for the Arduino?